src/garage/torch/algos/vpg.py

```diff
@@ -107,34 +123,28 @@ def _check_entropy_configuration(self, entropy_method, center_adv,
 
     def train_once(self, itr, paths):
         """Perform one step of policy optimization."""
-        samples_data = self.process_samples(itr, paths)
+        valids, obs, actions, rewards, average_return = self.process_samples(
+            itr, paths)  # noqa: E501
 
-        self._optimize_policy(itr, paths)
-        self.baseline.fit(paths)
-
-        return samples_data['average_return']
+        loss = self._compute_loss(itr, paths, valids, obs, actions, rewards)
 
-    def _optimize_policy(self, itr, paths):
         self.optimizer.zero_grad()
+        # using a negative because optimizers use gradient descent,
+        # whilst we want gradient ascent.
+        (-loss).mean().backward()
+        self.optimizer.step()
 
-        policy_entropies = torch.stack([
-            self._add_padding(
-                self.policy.get_entropy(torch.Tensor(path['observations'])),
-                self.max_path_length) for path in paths
-        ])
+        self.baseline.fit(paths)
+        return average_return
 
-        valids = [len(path['actions']) for path in paths]
+    def _compute_loss(self, itr, paths, valids, obs, actions, rewards):
+        policy_entropies = self.policy.get_entropy(obs)
```

Comment:
this doesn't seem to obey `stop_entropy_gradient`

Reply:
changed the code to use `_get_policy_entropy`